src/go/rdctl/pkg/factoryreset/delete_data.go

```diff
@@ -0,0 +1,307 @@
+	if err := unregisterWSL(); err != nil {
+		return err
+	}
+	if err := deleteWindowsData(!removeKubernetesCache, "rancher-desktop"); err != nil {
+		return err
+	}
+	return clearDockerContext()
+}
+
+// Most of the errors in this function are reported, but we continue to try to delete things,
+// because there isn't really a dependency graph here.
+// For example, if we can't delete the Lima VM, that doesn't mean we can't remove docker files
+// or pull the path settings out of the shell profile files.
+func deleteLinuxLikeData(homeDir string, altAppHomePath string, configHomePath string, pathList []string) error {
+	if err := deleteLimaVM(); err != nil {
+		logrus.Errorf("Error trying to delete the Lima VM: %s\n", err)
+	}
+	for _, currentPath := range pathList {
+		if err := os.RemoveAll(currentPath); err != nil {
+			logrus.Errorf("Error trying to remove %s: %w", currentPath, err)
+		}
+	}
+	if err := clearDockerContext(); err != nil {
+		logrus.Errorf("Error trying to clear the docker context %s", err)
+	}
+	if err := removeDockerCliPlugins(altAppHomePath); err != nil {
+		logrus.Errorf("Error trying to remove docker plugins %s", err)
+	}
+	rawPaths := []string{
+		".bashrc",
+		".bash_profile",
+		".bash_login",
+		".profile",
+		".zshrc",
+		".cshrc",
+		".tshrc",
+	}
+	for i, s := range rawPaths {
+		rawPaths[i] = path.Join(homeDir, s)
+	}
+	rawPaths = append(rawPaths, path.Join(configHomePath, "fish", "config.fish"))
+
+	return removePathManagement(rawPaths)
+}
+
+func deleteLimaVM() error {
+	if err := directories.SetupLimaHome(); err != nil {
+		return err
+	}
+	execPath, err := os.Executable()
+	if err != nil {
+		return err
+	}
+	execPath, err = filepath.EvalSymlinks(execPath)
+	if err != nil {
+		return err
+	}
+	limactl := path.Join(path.Dir(path.Dir(execPath)), "lima", "bin", "limactl")
+	return exec.Command(limactl, "delete", "-f", "0").Run()
+}
+
+func removeDockerCliPlugins(altAppHomePath string) error {
+	cliPluginsDir := path.Join(dockerconfig.Dir(), "cli-plugins")
+	entries, err := os.ReadDir(cliPluginsDir)
+	if err != nil {
+		if errors.Is(err, syscall.ENOENT) {
+			// Nothing left to do here, since there is no cli-plugins dir
+			return nil
+		}
+		return err
+	}
+	for _, entry := range entries {
+		if entry.Type()&os.ModeSymlink != os.ModeSymlink {
+			continue
+		}
+		fullPathName := path.Join(cliPluginsDir, entry.Name())
+		target, err := os.Readlink(fullPathName)
+		if err != nil {
+			logrus.Errorf("Failed to follow the symbolic link for file %s: error: %s\n", fullPathName, err)
+			continue
+		}
+		if strings.HasPrefix(target, path.Join(altAppHomePath, "bin")+"/") {
+			os.Remove(fullPathName)
+		}
+	}
+	return nil
+}
+
+func removePathManagement(dotFiles []string) error {
+	for _, dotFile := range dotFiles {
+		byteContents, err := os.ReadFile(dotFile)
+		if err != nil {
+			if errors.Is(err, syscall.ENOENT) {
+				// Nothing left to do here, since there is no cli-plugins dir
```

Comment:
Incorrect comment — this is presumably talking about the various dotfiles.

Reply:
Yeah, copy/paste or code-shift error. As for the previous comment, I'm reverting back to using `Linux` to name the most popular Unix-type system out there these days, and the only one we currently support.